modules/tinymce/src/plugins/table/test/ts/browser/ui/TableValignButtonsTest.ts

```diff
@@ -0,0 +1,154 @@
+import { Assertions, Keys, Waiter } from '@ephox/agar';
+import { afterEach, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/mcagar';
+import { SugarBody, SugarElement } from '@ephox/sugar';
+import Editor from 'tinymce/core/api/Editor';
+import { EditorEvent } from 'tinymce/core/api/util/EventDispatcher';
+import { TableModifiedEvent } from 'tinymce/plugins/table/api/Events';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.plugins.table.ui.TableValignButtonsTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'table',
+    indent: false,
+    toolbar: 'tablecellvalign',
+    base_url: '/project/tinymce/js/tinymce',
+    setup: (editor: Editor) => {
+      editor.on('tablemodified', logEvent);
+    }
+  }, [ Plugin, Theme ], true);
+
+  let events: Array<EditorEvent<TableModifiedEvent>> = [];
+  const logEvent = (event: EditorEvent<TableModifiedEvent>) => {
+    events.push(event);
+  };
+
+  afterEach(() => {
+    events = [];
+  });
+
+  const setEditorContentTableAndSelection = (editor: Editor) => {
+    editor.setContent(
+      '<table>' +
+        '<tbody>' +
+          '<tr>' +
+            '<td>Filler</td>' +
```

Comment:
Can we add a test for applying this over range selections (i.e. multiple cells)?